linkerd/app/gateway/src/tests.rs

```diff
@@ -146,4 +161,22 @@ impl Test {
         bg.await?;
         Ok(rsp)
     }
+
+    async fn run_default_profile(self) -> Result<http::Response<http::BoxBody>, Error> {
+        let target = self.target.clone();
+        let suffix = self.suffix.clone();
+
+        let allow = NameMatch::new(Some(dns::Suffix::from_str(suffix).unwrap()));
+        let profile = if allow.matches(target.name()) {
+            Some(support::profile::only(profiles::Profile {
+                addr: Some(target.clone().into()),
+                ..profiles::Profile::default()
+            }))
+        } else {
+            None
+        };
+
+        let resp = self.run(profile).await;
+        resp
+    }
```

Comment:
```suggestion
    async fn run_default_profile(self) -> Result<http::Response<http::BoxBody>, Error> {
        let allow = NameMatch::new(Some(dns::Suffix::from_str(self.suffix).unwrap()));
        let profile = if allow.matches(self.target.name()) {
            Some(support::profile::only(profiles::Profile {
                addr: Some(self.target.clone().into()),
                ..profiles::Profile::default()
            }))
        } else {
            None
        };

        let resp = self.run(profile).await;
        resp
    }
```

Clippy complains about the `suffix.clone()` -- suffix is just a `'static str` reference so it can be copied without cloning. Furthermore, it looks like the first `target.clone()` isn't needed since it's already cloned below.